### PR TITLE
Sphinx doc chapter 23

### PR DESCRIPTION
**Pull request by maximedenes**
Depends on #7037.

### CODE REVIEW

File: doc/sphinx/addendum/extraction.rst [hunk 1]
```diff
@@ -1,267 +1,247 @@
...
 
 Concerning Haskell, type-preserving optimizations are less useful
 because of laziness. We still make some optimizations, for example in
 order to produce more readable code.
 
-The type-preserving optimizations are controlled by the following \Coq\ options:
+The type-preserving optimizations are controlled by the following |Coq| options:
 
-\begin{description}
+.. cmd:: Unset Extraction Optimize.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extraction.rst]: Will be missing from the option index. Several same problems below.

File: doc/sphinx/addendum/extraction.rst [hunk 2]
```diff
@@ -1,267 +1,247 @@
...
 
-\item \optindex{Extraction Optimize} {\tt Unset Extraction Optimize.}
+   Default is ``Set``. This controls all type-preserving optimizations made on
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extraction.rst]: Default is on, ..., turn this option off would sound better. Same remark below.

File: doc/sphinx/addendum/extraction.rst [hunk 3]
```diff
@@ -1,267 +1,247 @@
...
+   inductive constructor, and the given `ident` are the names of
+   the concerned arguments. In fact, an argument can also be referred
+   by a number indicating its position, starting from 1.
+
+When an actual extraction takes place, an error is normally raised if the
+``Extraction Implicit`` declarations cannot be honored, that is
+if any of the implicited variables still occurs in the final code.
+This behavior can be relaxed via the following option:
+
+.. cmd:: Unset Extraction SafeImplicits.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extraction.rst]: Same remark wrt the option index.

File: doc/sphinx/addendum/extraction.rst [hunk 4]
```diff
@@ -273,348 +253,331 @@ a closed term, and of course the system cannot guess the program which
...
+.. cmd:: Extract Constant @qualid => @string.
+
+   Give an ML extraction for the given constant.
+   The `string` may be an identifier or a quoted string.
+
+.. cmd:: Extract Inlined Constant @qualid => @string.
+
+   Same as the previous one, except that the given ML terms will
+   be inlined everywhere instead of being declared via a ``let``.
+   Note: this command is sugar for an ``Extract Constant`` followed
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extraction.rst]: Note as a Sphinx equivalent.

File: doc/sphinx/addendum/extraction.rst [hunk 5]
```diff
@@ -273,348 +253,331 @@ a closed term, and of course the system cannot guess the program which
...
-\end{itemize}
-
-\Example
+types and constructors. For instance, the user may want to use the ML
+native boolean type instead of |Coq| one. The syntax is the following:
+
+.. cmd:: Extract Inductive @qualid => @string [ @string ... @string ].
+
+   Give an ML extraction for the given inductive type. You must specify
+   extractions for the type itself (first `string`) and all its
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extraction.rst]: Not `@string` instead?

File: doc/sphinx/addendum/extraction.rst [hunk 6]
```diff
@@ -273,348 +253,331 @@ a closed term, and of course the system cannot guess the program which
...
+have the correct number of arguments, etc. Of course, when launching
+manually some extracted function, you should apply it to arguments
+of the right shape (from the |Coq| point-of-view).
 
+More details about the correctness of the extracted programs can be 
+found in :cite:`Let02`.
 
+We have to say, though, that in most "realistic" programs, these
+problems do not occur. For example all the programs of Coq library are
+accepted by Caml type-checker without any ``Obj.magic`` (see examples below).
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extraction.rst]: OCaml (was like this in the LaTeX source too so this is not a regression)

**Review by Zimmi48 — COMMENTED** [review]
Not many problems in this chapter. The main one is the confusion between options and commands.

File: doc/sphinx/addendum/extraction.rst (hunk 5 above)
Comment (maximedenes) [reply doc/sphinx/addendum/extraction.rst]: No, the `@` has a special meaning and is not displayed.

**Comment by maximedenes** [comment]
Pipeline here: https://gitlab.com/maximedenes/coq/pipelines/19693273

Should be ready to go once CI says ok.

**Comment by maximedenes** [comment]
@silene Pipeline succeeded (modulo one spurious failure). Can you merge? (after #7037 is merged)

**Pull request merged**